File: tmdb-core/src/main/java/dev/reuss/tmdb/common/image/TvEpisodeImages.java
package dev.reuss.tmdb.common.image;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Image metadata for a TMDB TV episode.
 *
 * @param id     TMDB TV episode id
 * @param stills episode still images
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvEpisodeImages(
        int id,
        List<TmdbImage> stills
) implements TmdbModel {

    public TvEpisodeImages {
        stills = TmdbCollections.list(stills);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/common/image/TvEpisodeImages.kt
package dev.reuss.tmdb.common.image

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Image metadata for a TMDB TV episode.
 *
 * @property id     TMDB TV episode id
 * @property stills episode still images
 */
@JvmRecord
data class TvEpisodeImages(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("stills")
    val stills: List<TmdbImage> = emptyList(),
) : TmdbModel
